File: src/test/java/com/sesame/util/TextUtilsTest.java
package com.sesame.util;

import org.junit.Test;

import java.util.Optional;

import static org.junit.Assert.*;

public class TextUtilsTest {

    @Test
    public void shouldExtractValidFormat() {
        String commandLine = "@doclib " +
                "test" +
                "info" +
                "@enddoclib";

        Optional<String> result = TextUtils.extractDoclibTags(commandLine);

        assertTrue(result.isPresent());
        assertEquals(" testinfo", result.get());
    }

    @Test
    public void shouldNotExtractValidFormat() {
        String commandLine = "@doc " +
                "test" +
                "info" +
                "@enddoclib";

        Optional<String> result = TextUtils.extractDoclibTags(commandLine);

        assertFalse(result.isPresent());
    }

    @Test
    public void shouldNotExtractNullValue() {
        Optional<String> result = TextUtils.extractDoclibTags(null);
        assertFalse(result.isPresent());
    }

    @Test
    public void shouldNotExtractEmptyValue() {
        Optional<String> result = TextUtils.extractDoclibTags("");
        assertFalse(result.isPresent());
    }
}

File: src/test/kotlin/com/sesame/core/util/TextUtilsTest.kt
package com.sesame.core.util

import org.junit.Assert.*
import org.junit.Test

class TextUtilsTest {

    @Test
    fun `should extract valid format`() {
        val configDescription = "@doclib" +
                "test" +
                "info" +
                "@enddoclib"

        val result = TextUtils.extractDocLibTags(configDescription)

        assertTrue(result.isPresent)
        assertEquals("testinfo", result.get())

    }

    @Test
    fun `should not extract invalid format`() {
        val configDescription = "@doc " +
                "test" +
                "info" +
                "@enddoclib"

        val result = TextUtils.extractDocLibTags(configDescription)

        assertFalse(result.isPresent)

    }

    @Test
    fun `should not extract empty values`() {

        val result = TextUtils.extractDocLibTags("")

        assertFalse(result.isPresent)

    }


}